src/Castle.Windsor.Extensions.DependencyInjection/Scope/ExtensionContainerScopeAccessor.cs

```diff
@@ -23,11 +23,7 @@ internal class ExtensionContainerScopeAccessor : IScopeAccessor
 	{
 		public ILifetimeScope GetScope(CreationContext context)
 		{
-			if(ExtensionContainerScope.Current == null)
-			{
-				throw new InvalidOperationException("No scope available");
-			}
-			return ExtensionContainerScope.Current;
+			return ExtensionContainerScopeCache.Current ?? throw new InvalidOperationException("No scope available");
```

Comment:
`Current` already throws, no need for the check to throw here.

Reply:
done, and thanks